### ISSUE TITLE
Gradient reftest regression

### ISSUE THREAD

**Issue by glennw** [servo/webrender#1051]
The following Servo tests are broken by https://github.com/servo/webrender/pull/1046.
 
```
  ▶ FAIL [expected PASS] /_mozilla/css/border-image-linear-gradient.html
  └   → /_mozilla/css/border-image-linear-gradient.html e68bebc2dbcc53b852fd3b6e77485a9031fb5f35
/_mozilla/css/border-image-linear-gradient-ref.html 57d716ccc54e3f76d38f5babca226e3878731ca3
Testing e68bebc2dbcc53b852fd3b6e77485a9031fb5f35 == 57d716ccc54e3f76d38f5babca226e3878731ca3

  ▶ FAIL [expected PASS] /_mozilla/css/linear_gradients_parsing_a.html
  └   → /_mozilla/css/linear_gradients_parsing_a.html 95f9d6467048d6325dd24fe511c046a8a77d050e
/_mozilla/css/linear_gradients_parsing_ref.html 943c7b8a33e43d322e511d8ec6784868b781ac75
Testing 95f9d6467048d6325dd24fe511c046a8a77d050e == 943c7b8a33e43d322e511d8ec6784868b781ac75
```

We have another change landed in WR that I'd like to get into a WR update for Servo (may help with some of our intermittents). Is this likely to be a relatively simple fix, or should we revert for now?

cc @rlhunt @kvark

**Comment by rlhunt** [comment]
I will take a look at this as soon as I can.

**Comment by rlhunt** [comment]
The issue here is a few things:
1. One of the patches added a condition to when we use aligned gradients, but it is flipped backwards causing angle gradients to always be used
2. Angle gradients still have dithering enabled, even though aligned gradients had them disabled
3. There are reftest issues with dithering (should be fixed by #1045)

I will add a PR fixing the aligned gradient condition which fixed the issue for me.
